core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -690,11 +702,50 @@ private void bindVariables(
                 throw BadProtocolException.INSTANCE;
             }
             ensureData(lo, valueLen, msgLimit, j);
+            //infer type if needed
+            if (inferTypes) {
+                int pgType = inferParameterType(lo, valueLen);
```

Comment:
`inferParameterType()` is liable to return -1, which is not handled and not tested either.

Reply:
fixed